test/adapter/include_option_to_hash_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class IncludeOptionToHashTest < Minitest::Test
+      def test_nil
+        assert_equal(
+          {},
+          Adapter.include_option_to_hash(nil))
+      end
+
+      def test_empty_string
+        assert_equal(
+          {},
+          Adapter.include_option_to_hash(''))
+      end
+
+      def test_single_string
+        assert_equal(
+          { author: {} },
+          Adapter.include_option_to_hash('author'))
+      end
+
+      def test_multiple_strings
+        assert_equal(
+          { author: {}, comments: {} },
+          Adapter.include_option_to_hash('author,comments'))
+      end
+
+      def test_multiple_strings_with_space
+        assert_equal(
+          { author: {}, comments: {} },
+          Adapter.include_option_to_hash('author, comments'))
+      end
+
+      def test_nested_string
+        assert_equal(
+          { posts: { author: {} } },
+          Adapter.include_option_to_hash('posts.author'))
+      end
+
+      def test_multiple_nested_string
+        assert_equal(
+          { posts: { author: {}, comments: { author: {} } }, comments: {} },
+          Adapter.include_option_to_hash('posts.author,posts.comments.author,comments'))
+      end
+
+      def test_empty_array
+        assert_equal(
+          {},
+          Adapter.include_option_to_hash([]))
+      end
+
+      def test_simple_array
+        assert_equal(
+          { author: {}, comments: {} },
+          Adapter.include_option_to_hash([:comments, :author]))
+      end
+
+      def test_nested_array
+        assert_equal(
+          { posts: { author: {}, comments: { author: {} } }, comments: {} },
+          Adapter.include_option_to_hash([:comments, posts: [:author, comments: [:author]]]))
```

Comment:
for more complex data, having the method call as part of the assert_equal can make it more difficult to compare the expected with the input, and result.

for the sake of debugging, I like to do

    expected = ...
    parameter_stuff = ...
    actual = something(parameter_stuff)

    assert_equal(expected, actual)

that way, if something is going weird, you can easily throw a `binding.pry` in there without getting lost in minitest/rspec.